PracticeGit/test.c

```diff
@@ -0,0 +1,125 @@
+// test.c - Function that tests the program and main.
+// ------------------------------------------------------------------------------------------------
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include "Program.h"
+
+#define RED     "\033[1;31m"
+#define GREEN   "\033[1;32m"
+#define CYAN    "\033[1;36m"
+#define RESET   "\033[0m"
+
+/// <summary>Execute palindrome test cases </summary>
+void RunPalindromeTest ();
+
+/// <summary>Execute Integer Reversal test cases </summary>
+void RunIntegerReversalTest ();
+
+///  <summary>Get user input for Palindrome checker </summary>
+void GetPalindromeInput ();
+
+///  <summary>Get user input for Integer reversal </summary>
+void GetIntegerInput ();
+
+void RunPalindromeTest () {
+   const char* testStrings[] = { "Eva, can I stab bats in a cave?", "malaya&lam", "Hey!! World", "J", "No lemon, no melon","$$$",
+                                 "Her favorite number is 2", "    ra   c eca  r " };
+   int expectedResults[] = { PALINDROME, PALINDROME, NOT_PALINDROME, PALINDROME, PALINDROME,NOT_PALINDROME, NOT_PALINDROME, PALINDROME };
+   int numTestCases = sizeof (testStrings) / sizeof (testStrings[0]);
+   printf ("\n                           EXECUTE PALINDROME TEST CASES \n"
+      "+------------------------------------------------------+-----------------------+----------+ \n"
+      CYAN"|                      Test String                     |    Palindrome Check   |  Result  |\n"RESET
+      "+------------------------------------------------------+-----------------------+----------+\n");
+   for (int i = 0; i < numTestCases; i++) {
+      int result = IsPalindrome_s (testStrings[i]);
+      printf ("| %-52s | %-21s | %-19s |\n", testStrings[i], result == PALINDROME ? "Palindrome" : "Not a Palindrome", result == expectedResults[i] ? GREEN "Pass" RESET : RED "Fail" RESET);
+      printf ("+------------------------------------------------------+-----------------------+----------+\n");
+   }
+}
+
+void RunIntegerReversalTest () {
+   int testCases[] = { 78987, 159, -1234, 0, INT_MAX, INT_MIN, 22022022, -4567654 };
+   int expectedResults[] = { 78987, 951, -4321, 0, 0, 0, 22022022, -4567654 };
+   int numTestCases = sizeof (testCases) / sizeof (testCases[0]);
+   printf ("\n                   EXECUTE INTEGER REVERSAL TEST CASES\n"
+      "\n+-------------------+---------------------------+------------------------+----------+\n"
+      CYAN"| Original Integer  | Reversed Integer          | Palindrome Check       |  Result  |\n"RESET
+      "+-------------------+---------------------------+------------------------+----------+\n");
+   for (int i = 0; i < numTestCases; i++) {
+      int reversed, result = ReverseInteger (testCases[i], &reversed);
+      char reversedStr[12];    // Buffer for the reversed integer string
+      snprintf (reversedStr, sizeof (reversedStr), "%d", reversed);
+      char* isPalindrome = IsPalindrome_d (testCases[i]) ? "Palindrome" : "Not a palindrome";
```

Comment:
This serves no purpose in this test. You should have a separate test function for each functionality.